### PR TITLE
fix: prevent unnecessary database calls during API name updates

### LINKED ISSUE #812: Unnecessary database calls triggered by API-name update without actually changing the name
### Preliminary Checks

- [X] I have reviewed the documentation: https://unkey.dev/docs

- [X] I have searched for existing issues: https://github.com/unkeyed/unkey/issues

- [X] This issue is not a question, general help request, or anything other than a bug report directly related to Unkey. Please ask questions in our Discord community: https://unkey.dev/discord.

### Reproduction / Replay Link (Optional)

_No response_

### Issue Summary

When users attempt to update their API names without making any actual changes, the system is still triggering unnecessary database calls. The bug occurs during the update process, even when there is no modification made to the user's name.

### Steps to Reproduce

1. Log in to the application
2. Navigate to the API settings.
3. Click on the "Save" button to save the API name.

### Expected behavior

The system should recognize that there are no actual changes made to the API's name and should not initiate any database calls or updates.

### Other information

_No response_

### Screenshots

https://github.com/unkeyed/unkey/assets/87828904/b66db6c4-f2cd-4994-b2ae-070c0160ca37

### Version info

```markdown
- OS: Windows 11
- Node: v20.9.0
- npm: 10.1.0
```

### PR DESCRIPTION
**Pull request by ashrafchowdury**
## What does this PR do?

Fixes #812

Fixed the API name update issue, now users have to first change the API name to update it otherwise it will block to prevent unnecessary database calls.

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] Chore (refactoring code, technical debt, workflow improvements)
- [ ] Enhancement (small improvements)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## How should this be tested?

- Log in to the application
- Navigate to the API settings.
- Click on the API-name "Save" button to update the API name without changing it.

### Required

- [x] Filled out the "How to test" section in this PR
- [x] Read [Contributing Guide](./CONTRIBUTING.md)
- [x] Self-reviewed my own code
- [x] Commented on my code in hard-to-understand areas
- [x] Ran `pnpm build`
- [x] Ran `pnpm fmt`
- [x] Checked for warnings, there are none
- [x] Removed all `console.logs`
- [x] Merged the latest changes from main onto my branch with `git pull origin main`
- [x] My changes don't cause any responsiveness issues

### Appreciated

- [ ] If a UI change was made: Added a screen recording or screenshots to this PR
- [ ] Updated the Unkey Docs if changes were necessary

### CODE REVIEW

**Comment by chronark** [comment]
This can be frustrating, cause there's no feedback
can you please add a toast notification or disable the button?

**Review by chronark — CHANGES_REQUESTED** [review]
See comment above

File: apps/web/app/(authenticated)/(app)/app/apis/[apiId]/settings/update-api-name.tsx [hunk 1]
```diff
@@ -53,9 +53,14 @@ export const UpdateApiName: React.FC<Props> = ({ api }) => {
     },
   });
   async function onSubmit(values: z.infer<typeof formSchema>) {
+    if (!values.name) {
+      return toast.error("Please provide a valid name before saving.");
+    }
     if (values.name !== api.name) {
```
Comment (chronark) [inline apps/web/app/(authenticated)/(app)/app/apis/[apiId]/settings/update-api-name.tsx:59]: shouldn't this be the inverse?

if they're the same, we toast

**Review by chronark — CHANGES_REQUESTED** [review]

File: apps/web/app/(authenticated)/(app)/app/apis/[apiId]/settings/update-api-name.tsx (hunk 1 above)
Comment (ashrafchowdury) [reply apps/web/app/(authenticated)/(app)/app/apis/[apiId]/settings/update-api-name.tsx:59]: Should I add the condition something like this.

```
if (values.name !== api.name || !values.name) {
   return toast.error("Please provide a valid name before saving.");
}
```

Comment (chronark) [reply apps/web/app/(authenticated)/(app)/app/apis/[apiId]/settings/update-api-name.tsx:59]: No, if `values.name !== api.name` is true, then the new and old name are different, why are we toasting an error in this case?

Comment (ashrafchowdury) [reply apps/web/app/(authenticated)/(app)/app/apis/[apiId]/settings/update-api-name.tsx]: Ohh, my bed.

**Review by chronark — APPROVED** [review]

**Comment by chronark** [comment]
please sign the cla

**Comment by ashrafchowdury** [comment]
> please sign the cla

Signed it 🙌

**Pull request merged**